extensions/wikia/Email/EmailController.class.php

```diff
@@ -526,4 +530,106 @@ public function assertUserNotBlocked() {
 			throw new Check( 'User is blocked from taking this action' );
 		}
 	}
+
+	/**
+	 * Get the form field for this email to be used on Special:SendEmail
+	 * @return array
+	 */
+	public static function getAdminForm() {
+		return array_merge_recursive(
+			self::getBaseFormFields(),
+			static::getEmailSpecificFormFields()
+		);
+	}
+
+	/**
+	 * Get the common form fields used by all emails on Special:SendEmail.
+	 * @return array
+	 */
+	private static function getBaseFormFields() {
+		$baseForm = [
+			'inputs' => [
+				[
+					'type' => 'hidden',
+					'name' => 'token',
+					'value' => \F::app()->wg->User->getEditToken()
+				],
+				[
+					'type' => 'hidden',
+					'name' => 'emailController',
+					'value' => get_called_class()
+				],
+				[
+					'type' => 'hidden',
+					'name' => 'fromAddress',
+					'value' => \F::app()->wg->PasswordSender
+				],
+				[
+					'type' => 'hidden',
+					'name' => 'fromName',
+					'value' => \F::app()->wg->PasswordSenderName
+				],
+				[
+					'type' => 'text',
+					'name' => 'targetUser',
+					'label' => 'Target User',
+					'tooltip' => 'User to send the email to',
+					'value' => \F::app()->wg->User->getName()
```

Comment:
Ah, good to know. This is a special page which is intended only for staff to use so I don't think it would come up to much, but it's easy enough to just escape the name. I'll do that.